autoload/asyncomplete/sources/flow.vim

```diff
@@ -66,6 +70,31 @@ function! asyncomplete#sources#flow#get_source_options(opts)
        \ })
 endfunction
 
+let s:flowbin_path_by_dir = {} " dir: <path to flow>
+
+function! s:resolve_flowbin_path(file, fallback)
+    let l:dir = fnamemodify(a:file, ':h')
+    if !has_key(s:flowbin_path_by_dir, l:dir)
+        let l:node_dir = s:find_nearest_directory(a:file, 'node_modules')
+        if filereadable(l:node_dir . '/.bin/flow')
+            let s:flowbin_path_by_dir[l:dir] = l:node_dir . './.bin/flow'
+        else
+            let s:flowbin_path_by_dir[l:dir] = a:fallback
+        endif
+    endif
+    return s:flowbin_path_by_dir[l:dir]
+endfunction
+
+function! s:find_nearest_directory(filename, directoryname) abort
```

Comment:
This is useful for other sources too. Can you send the PR in asyncomplete repo?

Here is what you need to do. Create a new file called `autoload/asyncomplete/utils.vim` and rename the function to `asyncomplete#utils#find_nearest_directory`.

Note: method prefix should be path so make sure to follow the exact instructions.